pkg/venafi/firefly/connector.go

```diff
@@ -114,6 +114,11 @@ func (c *Connector) Authorize(auth *endpoint.Authentication) (token *oauth2.Toke
 		return config.PasswordCredentialsToken(context.Background(), auth.User, auth.Password)
 	}
 
+	// if it's a device flow grant
+	if auth.IdentityProvider.DeviceURL != "" {
+		return c.getDeviceAccessToken(auth)
+	}
+
 	// if it's a client credentials flow grant
 	if auth.ClientSecret != "" {
```

Comment:
ClientSecret should be the first authorization method we check. As user/password usage is discouraged and device is a niche use case.

Reply:
Level priority:
- ClientSecret
- Device
- User/Password

Reply:
done